recipes/util-linux-libuuid/all/conanfile.py

```diff
@@ -63,48 +65,64 @@ def _minimum_compiler_version(self, compiler, build_type):
         }
         return min_version.get(str(compiler), {}).get(str(build_type), "0")
 
+    def requirements(self):
+        if self.settings.os == "Macos":
+            # Required because libintl.{a,dylib} is not distributed via libc on Macos
+            self.requires("libgettext/0.22")
+
     def validate(self):
         min_version = self._minimum_compiler_version(self.settings.compiler, self.settings.build_type)
         if Version(self.settings.compiler.version) < min_version:
             raise ConanInvalidConfiguration(f"{self.settings.compiler} {self.settings.compiler.version} does not meet the minimum version requirement of version {min_version}")
         if self.settings.os == "Windows":
             raise ConanInvalidConfiguration(f"{self.ref} is not supported on Windows")
 
-    def requirements(self):
-        if self.settings.os == "Macos":
-            # Required because libintl.{a,dylib} is not distributed via libc on Macos
-            self.requires("libgettext/0.21")
+    def build_requirements(self):
+        self.tool_requires("bison/3.8.2")
+        self.tool_requires("meson/1.2.2")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/2.0.3")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
-        tc = AutotoolsToolchain(self)
-        tc.configure_args.append("--disable-all-programs")
-        tc.configure_args.append("--enable-libuuid")
+        tc = MesonToolchain(self)
+        tc.project_options["auto_features"] = "disabled"
+        tc.project_options["build-libuuid"] = "enabled"
         if self._has_sys_file_header:
-            tc.extra_defines.append("HAVE_SYS_FILE_H")
+            tc.preprocessor_definitions["HAVE_SYS_FILE_H"] = "1"
         if "x86" in self.settings.arch:
-            tc.extra_cflags.append("-mstackrealign")
+            tc.c_args.append("-mstackrealign")
         tc.generate()
+        pkg_config_deps = PkgConfigDeps(self)
+        pkg_config_deps.generate()
+        virtual_build_env = VirtualBuildEnv(self)
+        virtual_build_env.generate()
+        virtual_run_env = VirtualRunEnv(self)
+        virtual_run_env.generate()
```

Comment:
@jwillikers why do we need both virtual env?

Reply:
I had it there for the `self.requires("libgettext/0.22")`.